simple_network_sim/loaders.py

```diff
@@ -173,28 +174,25 @@ def readInfectiousStates(infectious_states: pd.DataFrame) -> List[Compartment]:
     return list(infectious_states.Compartment)
 
 
-def readInfectionProbability(df: pd.DataFrame) -> Dict[int, float]:
+def readInfectionProbability(df: pd.DataFrame) -> Dict[datetime.date, float]:
     """
     Transforms the dataframe from the data API into a dict usable inside the model
 
     :param df: a timeseries of infection probabilities
     :return: a timeseries of infection probabilities in the dict format
     """
-    probs: Dict[int, float] = {}
-    has_time_zero = False
+    assert len(df) != 0, "Dataframe must be non empty"
+
+    probs: Dict[datetime.date, float] = {}
     for row in df.to_dict(orient="row"):
-        time = int(row["Time"])
-        if time == 0:
-            has_time_zero = True
-        elif time < 0:
-            raise ValueError("can't have negative time")
+        if not isinstance(row["Date"], str):
+            raise ValueError("Date must be string")
+
+        date = datetime.datetime.strptime(row["Date"], '%Y-%m-%d').date()
```

Comment:
We could do with pandas operations, but this is much more flexible:
![image](https://user-images.githubusercontent.com/24376790/88074867-e874d280-cb6f-11ea-8b03-3e792c8a2bf0.png)

And we expose ourselves to issues in date formatting (users inputting dates in US vs European format), which I explicitly wanted to avoid by using the datetime format YYYY-MM-DD

Reply:
yeah, I had that concern too. I think it's better to do it like you did.